### PR TITLE
Refactoring of instruction sending logic

### PR DESCRIPTION
**Pull request by abuijze**
Each channel had the instruction dispatch and acknowledgement logic implemented for itself. This PR consolidates the logic and moves it to the shared parent class.

This PR has been built on top of #41.

### CODE REVIEW

File: src/main/java/io/axoniq/axonserver/connector/impl/AbstractAxonServerChannel.java [hunk 1]
```diff
@@ -16,45 +16,95 @@
...
+import java.util.function.Function;
+
+import static io.axoniq.axonserver.connector.impl.ObjectUtils.doIfNotNull;
+import static io.axoniq.axonserver.connector.impl.ObjectUtils.hasLength;
 
 /**
  * Abstract class representing a channel with AxonServer.
  */
-public abstract class AbstractAxonServerChannel {
+public abstract class AbstractAxonServerChannel<OUT> {
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/impl/AbstractAxonServerChannel.java:43]: The generic `OUT` should be specified in the javadoc I think.

File: src/main/java/io/axoniq/axonserver/connector/impl/AbstractAxonServerChannel.java [hunk 2]
```diff
@@ -16,45 +16,95 @@
...
-    protected AbstractAxonServerChannel(ScheduledExecutorService executor,
+    protected AbstractAxonServerChannel(ClientIdentification clientIdentification,
+                                        ScheduledExecutorService executor,
                                         AxonServerManagedChannel axonServerManagedChannel) {
+        this.clientIdentification = clientIdentification;
         this.executor = executor;
         this.channel = axonServerManagedChannel;
     }
 
+    protected CompletableFuture<Void> sendInstruction(OUT instruction, Function<OUT, String> instructionIdProvider, StreamObserver<OUT> outboundStream) {
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/impl/AbstractAxonServerChannel.java:66]: This method deserves a little javadoc

File: src/main/java/io/axoniq/axonserver/connector/impl/AbstractAxonServerChannel.java [hunk 3]
```diff
@@ -16,45 +16,95 @@
...
+                .orElse(() -> ack.completeExceptionally(new AxonServerException(ErrorCategory.INSTRUCTION_ACK_ERROR,
+                                                                                "Unable to send instruction: no connection to AxonServer",
+                                                                                clientIdentification.getClientId())));
+        if (!hasLength(instructionId)) {
+            ack.complete(null);
+        }
+        return ack;
+    }
+
+    protected void processAck(InstructionAck ack) {
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/impl/AbstractAxonServerChannel.java:83]: This method deserves a little javadoc

**Review by smcvb — APPROVED** [review]
Looks good to me :+1:, except for the missing javadoc. Trusting that'll be provided, hence approving preemptively.

**Review by m1l4n54v1c — APPROVED** [review]
LGTM

**Pull request merged**